src/control/server/mgmt_pool.go

```diff
@@ -657,6 +657,10 @@ func (svc *mgmtSvc) PoolDestroy(parent context.Context, req *mgmtpb.PoolDestroyR
 		return nil, err
 	}
 
+	return svc.poolDestroyNoLeaderCheck(parent, req)
+}
+
+func (svc *mgmtSvc) poolDestroyNoLeaderCheck(parent context.Context, req *mgmtpb.PoolDestroyReq) (*mgmtpb.PoolDestroyResp, error) {
```

Comment:
minor nit: We already have a convention of "inner" implementations that are called after whatever RPC preprocessing needs to happen (e.g. leader check, etc), so this could just be `poolDestroy()`. I don't object strongly to this being more explicit, but I can imagine that eventually the reasons for it being an inner helper will be for more than just the skipped leader check.

Reply:
I went back and forth with myself about the naming. I'll shorten it up if any other changes end up being needed, or if I need to merge master again.